### DISCUSSION TITLE
add tailwindcss cdn

### DISCUSSION THREAD

**Discussion by gracie2109** [eleith/emailjs#346] [questions] [answered]
Hi, can i add tailwindcss cdn in mail template? please answer me

**Comment by JeremiahRanen7** [comment, marked as answer]
> The answer is **YES**

You can definitely add a tailwindcss cdn in mail template , but :
 
- **Avoid Advanced Tailwind Features:** Some advanced features like **pseudo-classes** *(:hover, :focus)*, custom properties, and complex layout utilities may not work in emails.
- **Use Basic Tailwind Classes:** Stick to basic styling classes for padding, margins, colors, and typography to maximize compatibility.

**Comment by gracie2109** [comment]
hi, i'm a newbie and i can't find html file to add. can u create some mini
demo for me


Vào Th 7, 13 thg 7, 2024 lúc 20:33 Jeremiah Ranen R <
***@***.***> đã viết:

**Comment by eleith** [comment]
there is an example on this project README on how to send HTML files: https://github.com/eleith/emailjs?tab=readme-ov-file#example-usage---html-emails-and-attachments

as long as you have a valid HTML file, it should work. as @JeremiahRanen7 advises, just because your HTML file renders correctly, it is not gauranteed to render in an email client and thus you'll want to avoid using certain CSS classes.

you can use https://www.caniemail.com/ to learn what css/html is compatible with the email clients you want to focus on.
